src/runners/cairoRunner.ts

```diff
@@ -0,0 +1,175 @@
+import { MaybeRelocatable, Relocatable } from 'primitives/relocatable';
+import { Uint32, UnsignedInteger } from 'primitives/uint';
+import { Result } from 'result/result';
+import { Program } from 'vm/program';
+import { VirtualMachine } from 'vm/virtualMachine';
+
+export class CairoRunner {
+  private program: Program;
+  private vm: VirtualMachine;
+  private programBase: Relocatable;
+  private executionBase: Relocatable;
+  private initialPc: Relocatable;
+  private initialAp: Relocatable;
+  private initialFp: Relocatable;
+  private finalPc: Relocatable;
+  private mainOffset: Uint32;
+
+  constructor(program: Program) {
+    this.program = program;
+    const mainIdentifier = program.identifiers.get('__main__.main');
+    let mainOffset = 0 as Uint32;
+    if (mainIdentifier !== undefined && mainIdentifier.pc !== undefined) {
+      const { value: offset, error } = UnsignedInteger.toUint32(
+        mainIdentifier.pc
+      );
+      if (error !== undefined) {
+        throw new Error('Invalid main offset');
+      }
+      mainOffset = offset;
+    }
+
+    this.vm = new VirtualMachine();
+    this.programBase = new Relocatable(0, 0);
+    this.executionBase = new Relocatable(0, 0);
+    this.initialPc = new Relocatable(0, 0);
+    this.initialAp = new Relocatable(0, 0);
+    this.initialFp = new Relocatable(0, 0);
+    this.finalPc = new Relocatable(0, 0);
+    this.mainOffset = mainOffset;
+  }
+
+  // Initialize the runner with the program and stack.
+  initialize(): Result<Relocatable> {
+    this.initializeSegments();
+    const result = this.initializeMainEntrypoint();
+    this.initializeVm();
```

Comment:
this is confusing because we get as intermediary result `result` which is eventually returned. Why not 

```
    this.initializeSegments();
    this.initializeVm();
    return this.initializeMainEntrypoint();    
```

but I think that the content of these functions should be inlined here, they are small and it doesn't help readability

Reply:
after pulling the PR locally, I definitely think that you should inline everything and remove all the `initializeX` functions

Reply:
Mmmmm that might be a lot of code inlined into one function, worried this will not improve the lisibility nor the testing.